python-checks/src/main/java/org/sonar/python/checks/cdk/CdkUtils.java

```diff
@@ -215,17 +227,29 @@ private ExpressionFlow(SubscriptionContext ctx, Deque<Expression> locations) {
     }
 
     protected static ExpressionFlow build(SubscriptionContext ctx, Expression expression) {
+      return build(ctx, expression, null);
+    }
+
+    protected static ExpressionFlow build(SubscriptionContext ctx, Expression expression, @Nullable String argumentName) {
       Deque<Expression> locations = new LinkedList<>();
-      resolveLocations(expression, locations);
+      resolveLocations(ctx, expression, locations, argumentName);
       return new ExpressionFlow(ctx, locations);
     }
 
-    static void resolveLocations(Expression expression, Deque<Expression> locations) {
+    static void resolveLocations(SubscriptionContext ctx, Expression expression, Deque<Expression> locations, @Nullable String argumentName) {
```

Comment:
probably it is better to use LinkedHashSet here since there is a lot of `contains` checks. 
And then construct `LinkedList` out of `Set` results and pass it to the `ExpressionFlow` constructor.

Reply:
Sounds reasonable!